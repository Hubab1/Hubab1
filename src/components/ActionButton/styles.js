import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
