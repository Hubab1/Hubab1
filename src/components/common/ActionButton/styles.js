import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
