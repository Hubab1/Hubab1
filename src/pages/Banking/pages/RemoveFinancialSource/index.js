import RemoveFinancialSourcePage from './components/RemoveFinancialSourcePage/RemoveFinancialSourcePage';

export default RemoveFinancialSourcePage;
