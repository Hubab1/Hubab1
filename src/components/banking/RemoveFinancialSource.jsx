import React from 'react';
import styled from '@emotion/styled';

import { H1, H3, P, Bold } from 'assets/styles';
import captureRoute from 'app/captureRoute';
import { ROUTES, FINANCIAL_STREAM_ASSET, ALL_INCOME_OR_ASSET_TYPES } from 'app/constants';
import API from 'app/api';
import GenericFormMessage from 'components/common/GenericFormMessage';
import BankingContext from './BankingContext';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { prettyCurrency } from 'utils/misc';

const SkinnyH1 = styled(H1)`
    width: 70%;
`;

const SpacedH3 = styled(H3)`
    margin-top: 15px;
    margin-bottom: 22px;
`;

const Divider = styled.hr`
    border-style: none;
    border-bottom: 2px solid #EEEEEE;
    margin-bottom: 22px;
`;

export class RemoveFinancialSource extends React.Component {
    state = { errorSubmitting: false, financialSource: null, submitting: false }
    
    onSubmit = async () => {
        this.setState({submitting: true});
        try {
            await API.deleteFinancialSource(this.props.match.params.id);
        } catch {
            this.setState({submitting: false, errorSubmitting: true});
            return;
        }
        // eslint-disable-next-line
        this.context.refreshFinancialSources?.();
        this.setState({submitting: false});
        this.props.history.push(ROUTES.MANUAL_INCOME_VERIFICATION);
    };
    async componentDidMount () {
        this.fetchFinancialSource();
    }
    async fetchFinancialSource () {
        let data;
        try {
            data = await API.getFinancialSource(this.props.match.params.id);
        } catch (e) {
            return;
        }
        this.setState({financialSource: data});
    }
    componentDidUpdate (prevProps) {
        if (this.props.match.params.id !== prevProps.match.params.id) {
            this.fetchFinancialSource();
        }
    }

    render () {
        const financialSource = this.state.financialSource;
        if (!financialSource) return null;
        const isAsset = financialSource.stream_type === FINANCIAL_STREAM_ASSET;
        return (
            <>
                <SkinnyH1>Remove {isAsset ? 'Asset' : 'Income Source'}?</SkinnyH1>
                <SpacedH3>{ALL_INCOME_OR_ASSET_TYPES[financialSource.income_or_asset_type]?.label} - {prettyCurrency(financialSource.estimated_amount)}{isAsset ? '' : '/year'}</SpacedH3>
                <Divider />
                {this.state.errorSubmitting && (
                    <GenericFormMessage
                        type="error"
                        messages={['Oops! We had some trouble removing your financial source. Try again in a little bit.']}
                    />
                )}
                <Bold fontSize={18}>Are you sure you want to remove this {isAsset ? 'asset' : 'income source'}?</Bold><br/><br/>
                {!isAsset && <P>Removing this income source means that all uploaded files associated with it will be deleted and it will no longer count towards your total annual income.</P>}
                {isAsset && <P>Removing this asset means that all uploaded files associated with it will be deleted and it will no longer count towards your total asset balance.</P>}
                <ActionButton disabled={this.state.submitting} onClick={this.onSubmit} marginBottom={20} marginTop={100}>
                    Remove {isAsset ? 'Asset' : 'Income Source'}
                </ActionButton>
                <ActionButton onClick={()=>this.props.history.push(ROUTES.MANUAL_INCOME_VERIFICATION)} variant="outlined" marginBottom={20}>
                    Cancel
                </ActionButton>
            </>
        );
    }
}
RemoveFinancialSource.route = ROUTES.REMOVE_FINANCIAL_SOURCE;
RemoveFinancialSource.contextType = BankingContext;

export default captureRoute(RemoveFinancialSource);
