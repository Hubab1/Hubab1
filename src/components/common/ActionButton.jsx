import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}