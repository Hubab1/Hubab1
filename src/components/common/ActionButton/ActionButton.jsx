import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}