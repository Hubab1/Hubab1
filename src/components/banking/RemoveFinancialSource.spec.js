import React from 'react';
import { shallow } from 'enzyme';

import API from 'app/api';
import { RemoveFinancialSource } from './RemoveFinancialSource';
import ActionButton from 'components/common/ActionButton/ActionButton';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        match: {
            params: {
                id: '666'
            }
        },
        history: {
            push: jest.fn()
        }
    }
})

it('delete financial source on submit', async () => {
    API.deleteFinancialSource = jest.fn().mockReturnValue({});
    API.getFinancialSource = jest.fn().mockReturnValue({});
    const wrapper = await shallow(<RemoveFinancialSource {...defaultProps}/>);
    wrapper.find(ActionButton).at(0).simulate('click');
    expect(API.deleteFinancialSource).toHaveBeenCalledWith('666');
});
