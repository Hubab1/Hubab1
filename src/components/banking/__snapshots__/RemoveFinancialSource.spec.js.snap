// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when isAsset=false 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Remove 
    Income Source
    ?
  </ForwardRef(render)>
  <ForwardRef(render)>
     - 
    
    /year
  </ForwardRef(render)>
  <ForwardRef(render) />
  <div
    className="text-left"
  >
    <ForwardRef(render)
      fontSize={18}
    >
      Are you sure you want to remove this 
      income source
      ?
    </ForwardRef(render)>
    <br />
    <br />
    <ForwardRef(render)>
      Removing this income source means that all uploaded files associated with it will be deleted and it will no longer count towards your total annual income.
    </ForwardRef(render)>
  </div>
  <ActionButton
    color="primary"
    disabled={false}
    marginBottom={20}
    marginTop={100}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Remove 
    Income Source
  </ActionButton>
  <ActionButton
    color="primary"
    marginBottom={20}
    onClick={[Function]}
    successGreen={false}
    variant="outlined"
  >
    Cancel
  </ActionButton>
</React.Fragment>
`;

exports[`snapshot when isAsset=true 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Remove 
    Asset
    ?
  </ForwardRef(render)>
  <ForwardRef(render)>
     - 
    
    
  </ForwardRef(render)>
  <ForwardRef(render) />
  <div
    className="text-left"
  >
    <ForwardRef(render)
      fontSize={18}
    >
      Are you sure you want to remove this 
      asset
      ?
    </ForwardRef(render)>
    <br />
    <br />
    <ForwardRef(render)>
      Removing this asset means that all uploaded files associated with it will be deleted and it will no longer count towards your total asset balance.
    </ForwardRef(render)>
  </div>
  <ActionButton
    color="primary"
    disabled={false}
    marginBottom={20}
    marginTop={100}
    onClick={[Function]}
    successGreen={false}
    variant="contained"
  >
    Remove 
    Asset
  </ActionButton>
  <ActionButton
    color="primary"
    marginBottom={20}
    onClick={[Function]}
    successGreen={false}
    variant="outlined"
  >
    Cancel
  </ActionButton>
</React.Fragment>
`;
